### PR TITLE
Default to forward for tagged hashes

### PR DESCRIPTION
**Pull request by tcharding**
First three patches are preparation, improvements to the units tests in `sha256t`.

From the final patch:

    Displaying backward is an anomaly of Bitcoin Core's early days and the
    double SHA256 hash type. We should not let this unfortunate beast leak
    out into other places.
    
    Default to displaying forward when creating a new tagged hash and remove
    all the explicit attributes from `bitcoin` that just clutter the code.
 
This is an API break and may quietly break some users downstream - eventually we should stop doing that sort of thing.

### CODE REVIEW

**Comment by apoelstra** [comment]
Can you add some sort of migration instructions telling people they can add `#[hash_newtype(backward)]` to their structs if they want the old behavior?

I think this change is correct, even though it is theoretically an annoying break, because (I think) all tagged hashes are displayed forward.

**Comment by tcharding** [comment]
Done, as a separate patch.

**Review by apoelstra — DISMISSED** [review]
ACK f7e2b23b129199b00e959dde8e60a1a05df9b5f3

**Comment by tcharding** [comment]
This one can go in now, acked for two weeks.

File: hashes/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# unreleased
+
+* Change the default display direction of for tagged hashes to forwards. [#2707](https://github.com/rust-bitcoin/rust-bitcoin/pull/2707)
+
+  Note please this usage if you need to display backward:
+
+```
```
Comment (storopoli) [inline hashes/CHANGELOG.md:7]: Maybe syntax highlighting?

```suggestion
```rust
```

Comment (tcharding) [reply hashes/CHANGELOG.md:7]: Its in the changelog, who cares? Or if we do care, lets do it as a follow up. Waiting another two weeks just for this sucks balls.

Comment (apoelstra) [reply hashes/CHANGELOG.md:7]: Can you slip it in here?

I'll still ACK and (quickly) one-ACK merge. Agreed that a single nit about the CHANGELOG shouldn't reset the 2-week timer.

Comment (tcharding) [reply hashes/CHANGELOG.md]: Done, no other changes.

**Pull request merged**